web/src/main/java/se/inera/intyg/webcert/web/service/unansweredcommunication/UnansweredCommunicationServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+            throw new IllegalArgumentException(String.format(
+                "Request is missing parameters required to make request. patientIds '%s' maxDaysOfUnansweredCommunication '%s'",
+                patientIds,
+                maxDaysOfUnansweredCommunication));
+        }
+
+        final var arendenForPatients = arendeService.getArendenForPatientsWithTimestampAfterDate(
+            formatPatientIds(patientIds),
+            LocalDateTime.now().minusDays(maxDaysOfUnansweredCommunication)
+        );
+
+        if (arendenForPatients.isEmpty()) {
+            return new UnansweredCommunicationResponse(Collections.emptyMap());
+        }
+
+        return new UnansweredCommunicationResponse(createUnansweredCommunication(arendenForPatients));
+    }
+
+    private static List<String> formatPatientIds(List<String> patientIds) {
+        return patientIds.stream()
+            .map(patientId -> patientId.replace("-", ""))
+            .collect(Collectors.toList());
+    }
+
+    private Map<String, UnansweredQAs> createUnansweredCommunication(List<Arende> arenden) {
+        final var unansweredCommunication = new HashMap<String, UnansweredQAs>();
+        arenden.forEach(arende -> getUnansweredCommunication(arende, unansweredCommunication));
+        return unansweredCommunication;
+    }
+
+    private void getUnansweredCommunication(Arende arende, Map<String, UnansweredQAs> unansweredCommunication) {
+        final var status = convertStatus(arende);
+        if (isNotUnansweredCommunication(arende, status)) {
+            return;
+        }
+        final var unansweredQAs = getUnansweredQA(arende, unansweredCommunication);
+        if (KOMPLT.equals(arende.getAmne())) {
+            unansweredQAs.incrementComplement();
+        } else {
+            unansweredQAs.incrementOther();
+        }
+        unansweredCommunication.put(arende.getIntygsId(), unansweredQAs);
+    }
+
+    private static boolean isNotUnansweredCommunication(Arende arende, StatusType status) {
+        return status != OBESVARAD || arende.getAmne() == ArendeAmne.PAMINN || !arende.getSkickatAv().equals(FK);
+    }
+
+    private StatusType convertStatus(Arende arende) {
+        if (isAnswerFromExternal(arende.getStatus(), arende.getSvarPaId())) {
+            return BESVARAD;
+        }
+        switch (arende.getStatus()) {
+            case CLOSED:
+            case PENDING_EXTERNAL_ACTION:
+                return BESVARAD;
+            default:
+                return OBESVARAD;
+        }
+    }
+
+    private static boolean isAnswerFromExternal(Status status, String svarPaId) {
+        return status == Status.ANSWERED && svarPaId != null && !svarPaId.isEmpty();
+    }
+
+    private static UnansweredQAs getUnansweredQA(Arende arende, Map<String, UnansweredQAs> unansweredQAsMap) {
+        return unansweredQAsMap.containsKey(arende.getIntygsId()) ? unansweredQAsMap.get(arende.getIntygsId())
```

Comment:
You can do this more simply by using: 
`unansweredQAsMap.getOrDefault(arende.getIntygsId(), new UnansweredQAs(0, 0));`

Could also use default constructor as it will automatically set initial ints to 0 (new UnansweredQAs())

Reply:
Ahhh alright! Great tip! Thanks!